docs/interface.md

```diff
@@ -0,0 +1,56 @@
+# The Interface
+
+All interactions with the Exercism website are handled automatically. Test runners have the single responsibility of taking a solution, running its tests and returning a standarised output.
+
+## Execution
+
+- An test runner should provide an executable script. You can find more information in the [docker.md](https://github.com/exercism/automated-mentoring-support/blob/master/docs/docker.md) file.
+- The script will receive two parameters:
+  - The slug of the exercise (e.g. `two-fer`).
+  - A path to a directory containing the submitted file(s) (with a trailing slash).
+- The script must write a file to that directory named `report.json`
+
+## Output format
+
+The `report.json` file should be structured as followed:
+
+```json
+{
+  "status": "...",
+  "tests": [
+    {
+      "name": "Test that the thing works",
+      "status": "pass",
+      "expected": "42",
+      "actual": "123123",
+      "message": "Expected 42 but got 123123"
+    }
+  ]
+}
+```
+
+The following overall statuses are valid:
+- `pass`: All tests passed
+- `fail`: All tests failed
+- `error`: To be used when the tests did not run correctly (including if there are any skips)
+
+
+The following per-test statuses are valid:
+- `pass`: The test passed
+- `fail`: The test failed failed
+- `error`: The test errored
+- `skip`: The test was skipped
+
+Where possible, `expected` should show the output that was expected and `acutal` should show what was actually given.
+
+The `message` can be used to display human-readable error messages. Presume that whatever is written here will be displayed to the student.
+
+**We will provide a Ruby script that converts JUnit to this JSON output, which you can add as a final step of your Docker images, if you prefer to use an existing format.**
```

Comment:
Does this mean the image would have to include Ruby to use that script? In that case bash or some kind of standalone binary might help keeping the image small.

Reply:
To be decided. If we do require that, though, we can easily provide the relevant LOCs for the Dockerfile so that maintainers don't have to do more work.

Reply:
Ruby (or mri ruby as we use here on exercism) can not be compiled into a binary. So I agree that if this is a concern, ultimately we need something that can always run / can be compiled. I don't think this is an initial concern at all :)

Reply:
A small binary would indeed be preferrable. For the C# and F# tracks, I've built a prototype of my own logger (https://github.com/ErikSchierboom/Exercism.Logger), which was not hard to build. This approach might be something other tracks could also use.

Reply:
I'd also vote for a binary. It is much more portable.

Reply:
I'm not sure why portability matters? For both windows and linux we can provide a one-liner to the Dockerfiles to install Ruby and a single command/script to run for the transformation. I would also argue it's going to be easier for someone to hack and tweak the file to their needs in a scripting language than a compiled one (less tooling to have to work out, and I'd think Ruby/Python/etc had a higher level of familiarity / lower barrier to entry than something like Go for someone who's not familiar with either etc)

Reply:
Since you compared Go and Python -- both of which I have almost 4 years of experience with: Python is easy to learn, read and write, Go is even easier to learn, read and write.

Go can also be used as a scripting language. For that Go needs to be installed (as it would for python or ruby) and then a simple call e.g. to `go run main.go` would (compile and) run the program. Compilation is too fast to notice the delay on a small program as this would be.

Note: With things like [Yaegi](https://github.com/containous/yaegi) Go can be used even more like a scripting language (although that is still a very young endeavour and neither feature-complete nor considered API stable yet).

On the other hand more people know Python or Ruby. If this is going to be a thing of constant improvements and fixes (as @iHiD suggested it might be), then more people might be able to help. If that is not the case, a single executable is far easier to use than a ruby or python script.

Reply:
Helpful. Thanks @tehsphinx. FWIW, I find Go really hard to read (having never seen it other than reading the CLI code. But I guess that's a personal thing.

Reply:
@iHiD : Interesting. Then again, I find Ruby hard to read. I guess there is a big part of being used to it.